zipkin/src/test/java/zipkin/internal/DependencyLinkerTest.java

```diff
@@ -33,27 +34,40 @@ public void baseCase() {
   @Test
   public void linksSpans() {
     assertThat(new DependencyLinker().putTrace(TestObjects.TRACE).link()).containsExactly(
-        DependencyLink.create("web", "app", 1L),
-        DependencyLink.create("app", "db", 1L)
+      DependencyLink.create("web", "app", 1L),
+      DependencyLink.create("app", "db", 1L)
     );
   }
 
+  /***
+   * This is the test to show null pointer exception occuring
+   * when we encounter a span which has itself as a parent
+   * TestObjects.TRACEWITHSAMEIDANDSAMEPARENTID is a test object to simulate.
+   */
+
+  @Test(expected = NullPointerException.class)
+  public void linksSpansShouldHandleNullSpans() {
```

Comment:
rename this method. Obviously the data is very incorrect, also remember adjusting to not crash will hide the problem, punting it to other places.

a better method name would be linksSpans_skipsOnInvalidParentId (as opposed to crashing)